### PR TITLE
Fix MCP orderby validation to return actionable argument errors

### LINKED ISSUE #3810: [Bug]: MCP orderby takes incompatible shapes in read_records and aggregate_records, and the mismatch returns an opaque UnexpectedError
### What happened?

The two MCP DML tools accept `orderby` in mutually incompatible shapes. Passing one tool's shape to the other returns `UnexpectedError` with no indication of what was wrong, so the caller has no way to discover the correct form from the error.

**`aggregate_records` expects a bare direction string.** It sorts by the aggregated value.

| argument | result |
|---|---|
| `orderby: "desc"` | OK, groups sorted by aggregated value descending |
| `orderby: "asc"` | OK |
| `orderby: "COLUMN_NAME"` | `InvalidArguments: Argument 'orderby' must be either 'asc' or 'desc' when provided. Got: 'COLUMN_NAME'.` — clear and actionable |
| `orderby: ["desc"]` | `UnexpectedError: Unexpected error occurred in AggregateRecordsTool.` — opaque |

**`read_records` expects an array of column specifications.**

| argument | result |
|---|---|
| `orderby: ["COLUMN_NAME desc"]` | OK |
| `orderby: "COLUMN_NAME desc"` | `UnexpectedError: Unexpected error occurred in ReadRecordsTool.` — opaque |
| `orderby: "COLUMN_NAME"` | same |

So the shape that is correct in one tool is an opaque failure in the other, in both directions.

### Why this matters more for MCP than for REST

The consumer is a language model reading `describe_entities` output and the tool input schemas. Having learned `["COLUMN desc"]` from `read_records`, it will use that in `aggregate_records` and get a failure that names neither the parameter nor the expected form. There is no discovery path from the error back to the working call.

### Asks

1. Return an actionable message for the shape mismatch in both tools, as `aggregate_records` already does for the column-name case. Naming the expected shape would be enough.
2. Consider accepting both shapes, or aligning them.

### Two related observations

`aggregate_records` **silently ignores** `orderby` when `groupby` is absent, rather than rejecting it. A caller asking for sorted output gets unsorted output and a success response.

Grouped results are returned ordered by the aggregated value, and there is no way to order by the group key, because `orderby` accepts only a direction. For a `groupby` on a date or period column this means the caller cannot ask for chronological order and has to sort client-side. Given that filtering on a date-typed column is currently broken (#3760), `groupby` on the date is the only way to reach a single period, which makes the inability to order by that key more limiting than it would otherwise be.

### Version

2.1.3-rc

### What database are you using?

Azure SQL

### What hosting model are you using?

Azure Container Apps

### Which API approach are you accessing DAB through?

MCP

### PR DESCRIPTION
**Pull request by aaronburtle**
## Why make this change?

Closes https://github.com/Azure/data-api-builder/issues/3810

## What is this change?

- Addresses the opaque `UnexpectedError` responses reported in [#3810](https://github.com/Azure/data-api-builder/issues/3810).
- Validates `orderby` JSON types before accessing their values:
  - `read_records` requires an array of non-empty strings and validates every array member before metadata resolution.
  - `aggregate_records` requires a direction string, while preserving its existing null/blank handling.
- Invalid shapes now return `InvalidArguments` with the parameter name, expected format, and an example.
- Preserves the existing ordering contracts:
  - Read operations sort by one or more entity fields.
  - Grouped aggregates sort by the computed aggregate value, defaulting to descending order.
  - Scalar aggregate ordering remains ignored without grouping, preserving the behavior introduced by [#3294](https://github.com/Azure/data-api-builder/pull/3294).

## How was this tested?

- [x] Integration Tests 
- [x] Unit Tests

27 new error-validation regression cases failed against the original implementation and passed after the fix.

### Invalid scalar ordering for `read_records`

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "tools/call",
  "params": {
    "name": "read_records",
    "arguments": {
      "entity": "Book",
      "orderby": "id desc"
    }
  }
}
```

Previously returned `UnexpectedError`. Now returns `InvalidArguments` explaining that `orderby` must be an array of non-empty strings.

The supported form remains `"orderby": ["id desc"]`.

### Invalid array ordering for `aggregate_records`

```json
{
  "jsonrpc": "2.0",
  "id": 2,
  "method": "tools/call",
  "params": {
    "name": "aggregate_records",
    "arguments": {
      "entity": "Book",
      "function": "count",
      "groupby": ["title"],
      "orderby": ["desc"]
    }
  }
}
```

Previously returned `UnexpectedError`. Now returns `InvalidArguments` explaining that `orderby` must be a string containing `asc` or `desc`.

### CODE REVIEW

File: src/Service.Tests/Mcp/AggregateRecordsToolTests.cs [hunk 1]
```diff
@@ -495,6 +495,66 @@ public async Task AggregateRecords_GroupByDependencyViolation_ReturnsInvalidArgu
...
+        [DataTestMethod]
+        [DataRow("[\"desc\"]")]
+        [DataRow("[\"title desc\"]")]
+        [DataRow("[]")]
+        [DataRow("[null]")]
+        [DataRow("{}")]
+        [DataRow("123")]
+        [DataRow("true")]
+        [DataRow("false")]
+        public async Task AggregateRecords_InvalidOrderbyShape_ReturnsActionableInvalidArguments(string orderby)
```
Comment (souvikghosh04) [inline src/Service.Tests/Mcp/AggregateRecordsToolTests.cs:512]: Could we add a multi-field ordering integration test with duplicate values in the first sort column? Asserting the second column’s descending order would cover field precedence and the sort-expression construction changed in this PR.

**Review by souvikghosh04 — APPROVED** [review]

**Review by RubenCerna2079 — APPROVED** [review]
